### PR TITLE
Add basic text support to match Android [WIP]

### PR DESCRIPTION
**Pull request by ahtierney**
This request aims to provide basic text support to mirror support provided in the Android project as was addressed in PRs [343](https://github.com/airbnb/lottie-android/pull/343) and [353](https://github.com/airbnb/lottie-android/pull/353)

Goals:
- [ ] Support dynamically provided text (via LOTTextValueDelegate or similar)
- [x] Support AE JSON exported glyphs
- [ ] Support Stroke Width, Fill Color, Stroke Color and Tracking animations for parity with Android

### CODE REVIEW

**Comment by ahtierney** [comment]
@buba447 thanks for Lottie! We'd love to use it in a project that has text internationalization requirements and I was hoping to help get iOS somewhere near parity with Android. Before I got too deep into the rendering I wanted to post this to validate a few assumptions (mostly being that this would be a chunk of work you'd entertain at this point) and get any requirements from you as to direction. Thanks again!

**Comment by ahtierney** [comment]
I've added basic support for text animations with exported glyphs from AE following the work from Android [#343](https://github.com/airbnb/lottie-android/pull/343). This approach converts the text document info into the existing render models and arranges them appropriately.

hello

File: lottie-ios/Classes/AnimatableProperties/LOTKeyframe.m [hunk 1]
```diff
@@ -56,6 +56,44 @@ - (instancetype)initWithValue:(id)value {
...
 
+- (instancetype)initWithValue:(id)value forTime:(NSNumber*)keyframeTime {
+  self = [self initWithValue:value];
+  if (self) {
+    _keyframeTime = keyframeTime;
+  }
+  return self;
+}
+
+// TODO: check all nil guaruntees, ensure these two inits are good to go
```
Comment (Coeur) [inline lottie-ios/Classes/AnimatableProperties/LOTKeyframe.m]: small typo on `guaruntees`

File: lottie-ios/Classes/Extensions/LOTBezierPath.h [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (Coeur) [inline lottie-ios/Classes/Extensions/LOTBezierPath.h]: We don't need to add this empty line. :)

File: lottie-ios/Classes/Models/LOTShapeStroke.m [hunk 3]
```diff
@@ -12,13 +12,45 @@
...
+
+  self = [self init];
+  if (self) {
+    _keyname = keyname;
+    _color = colorFrames;
+    _width = widthFrames;
+    _opacity = opacityFrames;
+    _fillEnabled = false;
+    _dashOffset = nil;
+    // TODO: #? remove miter limit from LOTBezier as it's currenlty not used.
```
Comment (Coeur) [inline lottie-ios/Classes/Models/LOTShapeStroke.m]: small typo on `currenlty`

File: lottie-ios/Classes/Models/LOTText.m [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+  if (_animations.fillColor) {
+    _fillColorKeyframes = _animations.fillColor;
+  } else if (_properties.fontColor) {
+    LOTKeyframe *colorKeyframe = [[LOTKeyframe alloc] initWithColorValue:_properties.fontColor];
+    _fillColorKeyframes = [[LOTKeyframeGroup alloc] initWithKeyframes:@[colorKeyframe]];
+  } else {
+    _fillColorKeyframes = nil;
+  }
+
+  // layer frames are either visisble or not visible, iterate through the text frames and set this
```
Comment (Coeur) [inline lottie-ios/Classes/Models/LOTText.m]: typo on `visisble`

File: lottie-ios/Classes/Models/LOTText.m [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+      return preceedingCharactersWidths + ((tracking + inValue) * i);
+    }];
+
+    // map the tracking to x,y points
+    for (LOTKeyframe *frame in trackingKeyframes.keyframes) {
+      [frame formPointFromFloatWithYValue:0];
+    }
+
+    // create a transform node for this character scaling it by the font scale and positioning it
+    // in this frames character arrangment:
```
Comment (Coeur) [inline lottie-ios/Classes/Models/LOTText.m]: typo on `arrangment`

File: lottie-ios/Classes/Models/LOTText.m [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+    NSArray *fillItems = [character.shapes.items arrayByAddingObject:fillNode];
+
+    LOTShapeGroup *fillGroup = [[LOTShapeGroup alloc] initWithKeyname:character.characterString
+                                                               shapes:fillItems];
+
+    // arrange nodes based on "overfill":
+    NSArray *characterGroups;
+    // these are "fifo" because the render group always inserts at the 0 index so [stroke, fill]
+    // renders as fill, stroke in the layer heirarchy
```
Comment (Coeur) [inline lottie-ios/Classes/Models/LOTText.m]: typo on `heirarchy`

**Comment by ziyingjing** [comment]
text position Incorrect

**Comment by buba447** [comment]
Closing this PR. As of Lottie 3.0 (See https://github.com/airbnb/lottie-ios/pull/777), Lottie has been completely rewritten in swift. If you would like to continue using the Objective C version of Lottie we are accepting PR's on the official Objective-C branch https://github.com/airbnb/lottie-ios/tree/lottie/objectiveC

Cheers!

**Pull request closed without merging**